testing/steps/test_step_cloner_for_each_data_input.py

```diff
@@ -98,6 +101,30 @@ def test_step_cloner_should_set_train():
     assert not p.steps[1].is_train
 
 
+def test_step_cloner_should_save_sub_steps(tmpdir):
+    tape = TapeCallbackFunction()
+    p = StepClonerForEachDataInput(
+        Pipeline([
+            FitCallbackStep(tape),
+            MultiplyByN(2)
+        ]),
+        cache_folder_when_no_handle=tmpdir
+    )
+    data_inputs = _create_data((2, 2))
+    expected_outputs = _create_data((2, 2))
+    p, processed_outputs = p.fit_transform(data_inputs, expected_outputs)
+
+    p.save(ExecutionContext(tmpdir), full_dump=True)
```

Comment:
I am tempted to believe that the loading will bug if there is no self.steps_as_tuple in the cloner and that we'd need the refresh method of the truncable step... although you seem to have fixed saving.